### PR TITLE
fix: old password should not be equal to new password

### PR DESCRIPTION
**Pull request by jhasuraj01**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-7039 On cysync there isn't any check for old and new password](https://app.clickup.com/t/9002019994/PRF-7039)

File: packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx [hunk 1]
```diff
@@ -66,32 +66,23 @@ export const ChangePasswordDialogProvider: FC<
...
-        return;
-      }
-    }
+    let _password: string | null = null;
+    if (newPassword.length > 0) _password = newPassword;
+    if (confirmNewPassword.length > 0) _password = confirmNewPassword;
+
+    let _confirm: string | null = null;
+    if (confirmNewPassword.length > 0) _confirm = confirmNewPassword;
+    if (newPassword.length > 0) _confirm = newPassword;
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]:
```suggestion
    const _password = newPassword || confirmNewPassword || null;
    const _confirm = confirmNewPassword || newPassword || null;
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: .changeset/ten-comics-grow.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@cypherock/cysync-core': patch
+'@cypherock/cysync-desktop': patch
+---
+
+fix: old password should not be equal to new password
```
Comment (irshadCypherock) [inline .changeset/ten-comics-grow.md]:
```suggestion
old password should not be equal to new password
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx (hunk 1 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]: We need to check the string length as the type of `newPassword` and `confirmNewPassword` is `string` due to which they will never be a falsy value.

Comment (irshadCypherock) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]: Empty string is `false` is JS. Try out the suggested code.

Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]: ohh sorry! I got confused by `??` operator

Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/Password/ChangePassword/context/index.tsx]: Fixed

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**